test/gql_spec.js

```diff
@@ -1,20 +1,606 @@
+            // however Knex handles them. Dates are supported by GQL as a value type though.
+
+            // yes, this is intended to be 02-01, a date prior to all of the dates in our database
+            var query = gql.parse({created_at: {$gt: new Date('2016-02-01')}});
+            query.applyTo(knex('posts'))
+                .select()
+                .then(function (result) {
+                    result.length.should.eql(4);
+                    done();
+                });
+        });
+    });
+
+    describe('in clauses', function () {
+        it('should support in matches', function () {
+            gql.parse({created_at: ['2016-03-01', '2016-03-02']}).conditions
+                .should.eql({whereIn: ['created_at', ['2016-03-01', '2016-03-02']]});
+            gql.parse('created_at: [\'2016-03-01\', \'2016-03-02\']').conditions
+                .should.eql({whereIn: ['created_at', ['2016-03-01', '2016-03-02']]});
+        });
+
+        it('should support not in matches', function () {
+            gql.parse({$not: {created_at: ['2016-03-01', '2016-03-02']}}).conditions
+                .should.eql({whereNotIn: ['created_at', ['2016-03-01', '2016-03-02']]});
+            gql.parse('!created_at: [\'2016-03-01\', \'2016-03-02\']').conditions
+                .should.eql({whereNotIn: ['created_at', ['2016-03-01', '2016-03-02']]});
+        });
+    });
+
+    describe('invalid filters', function () {
+        it('should throw an error for a bad comparison operator (json api only)', function () {
+            (function () {
+                gql.parse({$bad: {created_at: ['2016-03-01', '2016-03-02']}});
+            }).should.throw();
+        });
+
+        it('should throw an error for an $or clause that does not have a value that\'s an array.', function () {
+            (function () {
+                gql.parse({$or: 'sample'});
+            }).should.throw();
+            (function () {
+                gql.parse(',sample');
+            }).should.throw();
+        });
+
+        it('should not throw an error for an $or clause that has a single object value', function () {
+            (function () {
+                gql.parse({$or: {name: 'sample'}});
+            }).should.not.throw();
+            (function () {
+                // this throws because it's not or'd with anything
+                gql.parse(',name:sample');
+            }).should.throw();
+        });
+    });
+
+    describe('aggregate queries', function () {
+        var usersAndPosts;
+        beforeEach(function () {
+            // the join and group by are not part of the query.
+            // the specify how the users and posts table relate.
+            // so we do that outside of the filters.
+            usersAndPosts = knex('users')
+                .join('posts', 'users.id', 'posts.author_id')
+                .groupBy('users.id');
+        });
+
+        it('should support $having', function (done) {
```

Comment:
Here you see an example of how orderBy and other knex functions can be used seamlessly.